src/MassTransit/Logging/EnabledDiagnosticSource.cs

```diff
@@ -0,0 +1,56 @@
+namespace MassTransit.Logging
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using Context;
+
+
+    public readonly struct EnabledDiagnosticSource
+    {
+        readonly DiagnosticSource _source;
+        readonly string _name;
+
+        public EnabledDiagnosticSource(DiagnosticSource source, string name)
+        {
+            _source = source;
+            _name = name;
+        }
+
+        public StartedActivity? StartActivity(object args = null)
+        {
+            Activity activity = new Activity(_name);
+
+            var startActivity = _source.StartActivity(activity, args);
+
+            var scope = LogContext.BeginScope();
+
+            return new StartedActivity(_source, startActivity, scope);
+        }
+    }
+
+
+    /// <summary>
+    /// This is an idea that might eventually be able to support passing value tuples and having them map to an argument type array
+    /// </summary>
+    readonly struct StartActivityArgument
+    {
+        public readonly string Key;
+        public readonly object Value;
+
+        public StartActivityArgument(string key, object value)
+        {
+            Key = key;
+            Value = value;
+        }
+
+        public static implicit operator StartActivityArgument((string, object) input)
+        {
+            return new StartActivityArgument();
```

Comment:
maybe I'm wrong but you forgot to pass arguments from `input`

Reply:
You aren't wrong, but the class isn't used (yet). It's an idea, it isn't real yet.

Reply:
I see you are trying over-OODesigning :-)

Reply:
I'm just thinking about convenience methods, and how tuples can be easily mapped to arguments without making it ridiculous.